pkg/resources/authentication_policy_acceptance_test.go

```diff
@@ -0,0 +1,78 @@
+package resources_test
+
+import (
+	"testing"
+
+	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+	"github.com/hashicorp/terraform-plugin-testing/config"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/tfversion"
+)
+
+func TestAcc_AuthenticationPolicy(t *testing.T) {
+	accName := acc.TestClient().Ids.Alpha()
```

Comment:
We need more tests, like 
- basic flow
  1. create without optional
  2. set optional fields
  1. import
  4. handle external change (one field is enough)
  3. unset optional fields
  1. import
 - renaming
 Please take a look at row_access_policy_acceptance_test.go - specifically TestAcc_RowAccessPolicy and TestAcc_RowAccessPolicy_Rename.
 
 We use custom generated models in there, but it can be tricky to set up, and I think you can use TF files + config map.

Reply:
@sfc-gh-jmichalak 
I have not fixed my local test setup and can now finalize the tests. However, one question, how to do "handle external change (one field is enough)"?

Reply:
Basically, you have use `PreConfig` (couldn't find a basic usage, but you can search by PreConfig and see how it's used) in the second step (or somewhere later on) and use raw SQL client to change the state of an object "by hand". Then, within the same test step you ensure that the values went back to the ones that are set in the Terraform configuration. For example, you have set comment in the Terraform configuration to "foo", then in the second step in PreConfig, you use `acc.TestClient().AuthPolicy.Alter(...)` to change this comment to "bar" and in the `Check` you assert comment to contain "foo", because we expect the provider to override incorrect value.